contrib/babelfishpg_tsql/src/procedures.c

```diff
@@ -1588,6 +1590,220 @@ create_xp_instance_regread_in_master_dbo_internal(PG_FUNCTION_ARGS)
+		rewrite_object_refs(stmt);
+
+		if (list_length(parsetree_list) != 1)
+			ereport(ERROR,
+					(errcode(ERRCODE_SYNTAX_ERROR),
+					errmsg("expected 1 statement but got %d statements after parsing", list_length(parsetree_list))));
+
+		stmt = ((RawStmt *) linitial(parsetree_list))->stmt;
+			
+		/* need to make a wrapper PlannedStmt */
+		wrapper = makeNode(PlannedStmt);
+		wrapper->commandType = CMD_UTILITY;
+		wrapper->canSetTag = false;
+		wrapper->utilityStmt = stmt;
+		wrapper->stmt_location = 0;
+		wrapper->stmt_len = len;
+
+		parsetree = wrapper->utilityStmt;
+
+		switch (nodeTag(parsetree))
+		{
+			case T_CreateExtensionStmt:
+			{
+				CreateExtensionStmt *crstmt = (CreateExtensionStmt *) parsetree;
+				DefElem    *d_schema = NULL;
+				ListCell   *lc;
+				char	   *schemaName = NULL;
+
+				for(int i = 0; i < allowed_extns_size; i++)
+				{
+					if(!(strcmp(crstmt->extname, allowed_extns[i])))
+					{
+						flag = true;
+					}
+				}
+
+				if(!flag)
+				{
+					ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					 		errmsg("'%s' extension creation is not supported", crstmt->extname)));
+				}
+
+				if (!superuser_arg(GetSessionUserId()) || !role_is_sa(GetSessionUserId()))
+				{
+					ereport(ERROR,
+						(errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
+	 					errmsg("permission denied to create extension")));
+				}
+
+				SetCurrentRoleId(GetSessionUserId(), false);
+				SetConfigOption("search_path",
+								new_path,
+								PGC_SUSET,
+								PGC_S_DATABASE_USER);
+
+				foreach(lc, crstmt->options)
+				{            
+					DefElem    *defel = (DefElem *) lfirst(lc);                                                                                                                                                                               
+					if (strcmp(defel->defname, "schema") == 0) 
+					{
+						d_schema = defel;
+						schemaName = defGetString(d_schema);
+					}
+					if (strcmp(defel->defname, "cascade") == 0)
+					{
+						ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+							errmsg("'cascade' is not yet supported in Babelfish")));
+					}
+				}
+
+				if(schemaName != NULL && !(is_shared_schema(schemaName)))
+				{
+					ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+							errmsg("'%s' is not a valid schema", schemaName)));
+				}
+
+				/* do this step */
+				ProcessUtility(wrapper,
+							postgresStmt,
+							false,
+							PROCESS_UTILITY_SUBCOMMAND,
+							NULL,
+							NULL,
+							None_Receiver,
+							NULL);
+
+				/* make sure later steps can see the object created here */
+					CommandCounterIncrement();
+				break;
+			}
+			case T_DropStmt:
+			{
+				DropStmt *drstmt = (DropStmt *) parsetree;
+
+				SetCurrentRoleId(GetSessionUserId(), false);
+
+				if(drstmt->behavior == DROP_CASCADE)
+				{
+					ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+								errmsg("'cascade' is not yet supported in Babelfish")));
+				}
```

Comment:
We should be checking that the object is extension or not. There no point setting sessionuser and checking the cascade if it's not drop extension statement. We need to throw unsupported error in case of other drop statements.
Can you please add a testcase where we do drop table ? In that our current implementation won't throw any error.